### PR TITLE
minion: auto-update the changelog

### PR DESCRIPTION
**Pull request by beyang**
https://github.com/sourcegraph/cody/assets/1646931/99259cc9-869f-4c8a-bbc3-da3638bd4245

Add the ability to automatically update the changelog given the feature description

## Test plan

Feature is experimental, behind an experimental feature flag

### CODE REVIEW

File: vscode/src/minion/blocks/plan-run-step.ts [hunk 1]
```diff
@@ -16,10 +29,75 @@ export async function runStep(
...
+    const prevUri = vscode.Uri.from({
+        scheme: 'git',
+        path: changelogUri.path,
+        query: JSON.stringify({ ref: '~', path: changelogUri.path }),
+    })
+    const curUri = vscode.Uri.from({
+        scheme: changelogUri.scheme,
+        path: changelogUri.path,
+    })
+    await vscode.commands.executeCommand('vscode.diff', prevUri, curUri, 'the diff')
```
Comment (abeatrix) [inline vscode/src/minion/blocks/plan-run-step.ts:100]: We should update this to use the Edit command here in a follow-up to get all the code lenses for undo, retry and show diff 😄 Something like:

```suggestion
    const document = await vscode.workspace.openTextDocument(changelogUri)
    await vscode.window.showTextDocument(document)

    await executeEdit({
        configuration: {
            document,
            instruction: prompts.changelogUser(clChunk, currentDate(), description), // Doesn't work
            mode: 'edit',
            range: new vscode.Range(0, 0, 50, 0),
            model: 'anthropic/claude-3-haiku-20240307',
        },
    } satisfies ExecuteEditArguments)
```

(The prompt needs to be updated for it to work)

**Review by abeatrix — APPROVED** [review]
lgtm!

image

File: vscode/src/minion/prompts.ts [hunk 2]
```diff
@@ -141,39 +141,40 @@ export function shouldAskHumanUser(description: string): MessageParam[] {
...
+Generate a patch to update the section you've chosen to modify. Your output should be in the following format:
+<linesToRemove>
+the lines that should be replaced, including distinctive lines preceeding and following
+</linesToRemove>
+<linesToInsert>
+the lines that should be substituted in place of the removed lines
+</linesToInsert>
+
+IMPORTANT RULES:
+- Do NOT generate any hyperlinks. Do not reference an issue or pull request numbers.
```
Comment (abeatrix) [inline vscode/src/minion/prompts.ts:158]: This seems to cause the LLM to remove links from the existing changelog entries:

image

Comment (beyang) [reply vscode/src/minion/prompts.ts:158]: Will fix in follow-up

File: vscode/src/minion/blocks/plan-run-step.ts (hunk 1 above)
Comment (beyang) [reply vscode/src/minion/blocks/plan-run-step.ts:100]: Will implement in follow-up

**Pull request merged**